docs/templates.md

```diff
@@ -601,3 +611,336 @@ access to the comments template tags and filters. The child template is
+a different user and therefore may have a different locale. The locale should be passed
+in to the rendering routines using the render context. Reinhardt can then use this
+locale to ask the resource bundle factory for a resource bundle for that locale, and
+then ask that resource bundle for translations to the strings it finds in the ``trans``
+and ``blocktrans`` tags. If no locale is given in the context, Reinhardt will use the
+default locale (if it is set up) or it will default to the text that is already in
+the templates.
+
+Here is an example of a route using stick that passes the locale in to Reinhardt
+in the render context:
+
+    app.get('/:page', function(request, page) {
+        return env.renderResponse(page, {
+            locale: request.session.data.locale
+        });
+    });
+
+If you use RingoJS's stick package, you can use the  ``locale`` middleware to
+set up the session variable ``request.session.data.locale`` automatically. You can
+also insert the user's locale into the session yourself when the user logs in based
+on their personal locale preference as saved in their user record in the data store.
+
+Customizing Templates Per Locale
+--------------------------------
+
+Third, we might need on occasion to customize templates per locale. For example, a dialog
+that asks a user to enter information about a contact entry should have different fields
+depending on the locale. In Japan, you would add a phonetic name field to the regular list
+of fields so that
+the user can tell you how the contact's family name is pronounced, and therefore how
+it sorts in the list of contacts. In France, you would have a CEDEX field in the set
+of address fields which will allow the user to add their contact's CEDEX information.
+(The CEDEX is a key which tells the main post office of big cities in France which
+sattelite office to send the mail to for sorting and delivery.)
+
+Because you have to put the locale setting into the context in order to do translation,
+this setting is also available to use in conditional statements. Here is
+an example:
+
+    {% if locale = "de-DE" %}
+        {% include "germandialog.html" %}
+    {% elif locale = "fr-FR" %}
+        {% include "germandialog.html" %}
+    {% else %}
+        {% include "englishdialog.html" %}
+    {% endif %}
+
+
+Globalized Filters
+------------------
+
+Currently, Reinhardt does not have any dependencies on other libraries that can
+give it the ability to do properly globalized filtering. As such, the filters that come
+with Reinhardt currently only support English/Western European behaviours. This may be fine
+for many sites, but if you need to support other locales, be careful when using
+the built-in filters. You might look for Ringo packages on RP that implement
+locale-sensitive filter plugins or implement them yourself.
+
+Just so that you are aware when you use them, the following filters perform
```

Comment:
Yes, I think we could even call it more explictely: "Problems with i18n in Reinhard" or so :) Not sure the detailed list of problematic filters makes sense I created https://github.com/orfon/reinhardt/issues/17 and https://github.com/orfon/reinhardt/issues/16

And IMO this is so much info that it needs its own i18n.md. Templates.md should just give a short overview - maybe the first paragraph - and then link to i18n.md. Like the other sections above it.